swap_meet/vendor.py

```diff
@@ -1,2 +1,78 @@
+from swap_meet.item import Item 
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_id(self, item_id): 
+        for item in self.inventory:
+            if item_id == item.id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.inventory.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        other_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        return True
+    
+    # Wave 4
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        first_item_vendor = self.inventory[0]
+        first_item_friends = other_vendor.inventory[0]
+        return self.swap_items(other_vendor, first_item_vendor, first_item_friends)
+
+    def get_by_category(self, category):
+        category_items = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_items.append(item)
+        return category_items
```

Comment:
⭐️
```suggestion
    def get_by_category(self, category):
        category_items = []
        
        for item in self.inventory:
            if item.get_category() == category:
                category_items.append(item)
                
        return category_items
```

Reply:
You could use some list comprehension here as well if you wanted to! I think that it would still be readable!